### PR TITLE
Add global error state and handle (un)expected errors

### LINKED ISSUE #277: Improve error pages design
We're working on an improved error page design ([see Figma](https://www.figma.com/design/zZlFr8tYiRyp4I26sh6eqp/Kiesraad---Abacus-optelsoftware?node-id=3933-59177&t=PubTloYfoXHhAha9-4)).

Once the design is finished this should be implemented in the frontend.
- The 404 Not Found page should be the handler for invalid frontend URLs 
- The 500 Internal Server Error page should be rendered for errors from the backend that we cannot easily recover from

### PR DESCRIPTION
**Pull request by marlonbaeten**
Also fixes #277

### CODE REVIEW

**Comment by marlonbaeten** [comment]
Note that this is currently a WIP - I plan to have this PR ready this week.

File: frontend/lib/api/useApiGetRequest.ts [hunk 1]
```diff
@@ -39,14 +47,13 @@ export function useApiGetRequest<T>(path: string): UseApiGetRequestReturn<T> {
           console.error("GET API error", e);
```
Comment (Lionqueen94) [inline frontend/lib/api/useApiGetRequest.ts]: Should there also be a setError here?

Comment (marlonbaeten) [reply frontend/lib/api/useApiGetRequest.ts]: My work in net ready for review yet ;-)

I was trying different approaches on this branch - but decided to split my changes in multiple PR's.

The flow of control / state in especially `PollingStationFormController` makes it a bit confusing on how and where to handle these errors. So first things first. I'll just introduce the error boundary in this PR to handle exceptions correctly and discuss refactoring some of the hooks (to improve readability / simplify control flow) next week.

**Comment by marlonbaeten** [comment]
Superseded by #471

**Pull request closed without merging**